rpc/resource/external/resource.go

```diff
@@ -0,0 +1,113 @@
+package external
```

Comment:
please add comments to explain the package. `external` is easy to misunderstand without full context